File: src/game/entity/enemies/FlyingEnemy.java
package game.entity.enemies;

import game.entity.blocks.DestroyableBlock;
import game.entity.models.*;
import game.models.Coordinates;
import game.ui.Paths;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class FlyingEnemy extends Enemy {
    public FlyingEnemy(Coordinates coordinates) {
        super(coordinates);
    }

    public String[] getFrontIcons() {
        return new String[]{
                Paths.getEnemiesFolder() + "/flying_enemy/aquila.png"
        };
    }

    @Override
    public String[] getLeftIcons() {
        return getFrontIcons();
    }

    @Override
    public String[] getBackIcons() {
        return getFrontIcons();
    }

    @Override
    public String[] getRightIcons() {
        return getFrontIcons();
    }

    @Override
    public Set<Class<? extends Entity>> getObstacles() {
        Set<Class<? extends Entity>> baseObstacles = new HashSet<>(super.getObstacles());
        baseObstacles.remove(DestroyableBlock.class);
        return baseObstacles;
    }


}

File: src/game/engine/world/entity/impl/enemies/npcs/FlyingEnemy.kt
package game.engine.world.entity.impl.enemies.npcs

import game.engine.world.entity.impl.blocks.DestroyableBlock
import game.engine.world.geo.Coordinates
import game.engine.world.entity.impl.models.Entity

/**
 * Enemy that can fly over destroyable blocks;
 */
abstract class FlyingEnemy : IntelligentEnemy {
    constructor() : super()
    constructor(coordinates: Coordinates?) : super(coordinates)
    constructor(id: Long) : super(id)

    override fun getObstacles(): Set<Class<out Entity>> = HashSet(super.getObstacles()).apply {
        remove(DestroyableBlock::class.java)
    }
}